lib/ui/platform_dispatcher.dart

```diff
@@ -2605,3 +2662,52 @@ class SemanticsActionEvent {
     );
   }
 }
+
+/// Signature for [PlatformDispatcher.onPlatformFocusChange].
+typedef ViewFocusChangeCallback = void Function(ViewFocusEvent);
+
+/// An event for the engine to communicate view focus changes back to the app.
+final class ViewFocusEvent {
+  /// Creates a [ViewFocusChange].
+  const ViewFocusEvent({
+    required this.viewId,
+    required this.focusState,
+  });
+
+  /// The id of the [FlutterView] that experienced a focus change.
+  final int viewId;
+
+  /// The current [ViewFocusState] of the [FlutterView] with the given [viewId].
+  final ViewFocusState viewFocusState;
+}
+
+/// Represents a given flutter view focus state.
+///
+/// If focus was lost, use [ViewFocusState.unfocused].
+/// If focus was gained without direction, use [ViewFocusState.focused].
+/// If focus was gained in forward direction, use [ViewFocusState.forwardFocused].
+/// If focus was gained in backwards direction, use [ViewFocusState.backwardsFocused].
+///
+/// Valid transitions, within a view, are:
+///
+/// - [ViewFocusState.unfocused] to [ViewFocusState.focused].
+/// - [ViewFocusState.unfocused] to [ViewFocusState.forwardFocused].
+/// - [ViewFocusState.unfocused] to [ViewFocusState.backwardsFocused].
+/// - [ViewFocusState.focused] to [ViewFocusState.unfocused].
+/// - [ViewFocusState.forwardFocused] to [ViewFocusState.unfocused].
+/// - [ViewFocusState.backwardsFocused] to [ViewFocusState.unfocused].
+///
+/// All others are invalid.
+enum ViewFocusState {
+  /// Used to indicate a given [FlutterView] lost focus.
+  unfocused,
+
+  /// Used to indicate a given [FlutterView] gained focus without direction.
+  focused,
+
+  /// Used to indicate a given [FlutterView] gained focus in forward direction.
```

Comment:
You could give these docs a little bit more details, e.g. that one way focus is moved in the forward direction is typically triggered by the user pressing the tab key. And that a typical responds to this kind of movement is to move the focus indication to the first focusable thing in the view.

Reply:
Done